src/core/surface/init.c

```diff
@@ -73,3 +73,13 @@ void grpc_shutdown(void) {
   }
   gpr_mu_unlock(&g_init_mu);
 }
+
+int grpc_is_initialized(void) {
+  int r;
+  gpr_once_init(&g_init, do_init);
```

Comment:
Yeah, fair enough. I'll wait until it goes green again :-)

On Thu, Mar 5, 2015 at 5:16 PM, Craig Tiller <notifications@github.com>
wrote:

> In src/core/surface/init.c
> <https://github.com/grpc/grpc/pull/964#discussion_r25918725>:
>
> > @@ -73,3 +73,13 @@ void grpc_shutdown(void) {
> >    }
> >    gpr_mu_unlock(&g_init_mu);
> >  }
> > +
> > +int grpc_is_initialized(void) {
> > +  int r;
> > +  gpr_once_init(&g_init, do_init);
>
> Pushed an update -- WDYT?
> … <#14beca7de80d4878_>
> On Thu, Mar 5, 2015 at 5:12 PM Nicolas Noble <notifications@github.com>
> wrote: In src/core/surface/init.c <
> https://github.com/grpc/grpc/pull/964#discussion_r25918566>: > @@ -73,3
> +73,13 @@ void grpc_shutdown(void) { > } > gpr_mu_unlock(&g_init_mu); > } >
> + > +int grpc_is_initialized(void) { > + int r; > + gpr_once_init(&g_init,
> do_init); I got slightly confused btw - I think "do_init" might be a
> misnomer. But I don't see any better name for it. — Reply to this email
> directly or view it on GitHub <
> https://github.com/grpc/grpc/pull/964/files#r25918566>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/964/files#r25918725>.
>